src/gui/editorwidgets/qgsvaluerelationconfigdlg.cpp

```diff
@@ -70,6 +71,23 @@ QVariantMap QgsValueRelationConfigDlg::config()
 void QgsValueRelationConfigDlg::setConfig( const QVariantMap &config )
 {
   QgsVectorLayer *lyr = QgsProject::instance()->mapLayer<QgsVectorLayer *>( config.value( QStringLiteral( "Layer" ) ).toString() );
+  // If layer is null, let's try to match it against name and provider
+  if ( ! lyr )
+  {
+    const auto name { config.value( QStringLiteral( "LayerName" ) ).toString() };
```

Comment:
Based on our previous discussion

```suggestion
    const QString name { config.value( QStringLiteral( "LayerName" ) ).toString() };
```